pkg/pgmodel/pgx.go

```diff
@@ -173,30 +179,66 @@ func NewPgxIngestorWithMetricCache(c *pgxpool.Pool, cache MetricCache) *DBIngest
 	return &DBIngestor{
 		db:    pi,
 		cache: bc,
-	}
+	}, nil
 }
 
 // NewPgxIngestor returns a new Ingestor that write to PostgreSQL using PGX
-func NewPgxIngestor(c *pgxpool.Pool) *DBIngestor {
+func NewPgxIngestor(c *pgxpool.Pool) (*DBIngestor, error) {
 	metrics, _ := bigcache.NewBigCache(DefaultCacheConfig())
 	cache := &MetricNameCache{metrics}
 	return NewPgxIngestorWithMetricCache(c, cache)
 }
 
-func newPgxInserter(conn pgxConn, cache MetricCache) *pgxInserter {
-	return &pgxInserter{
-		conn:             conn,
-		metricTableNames: cache,
+func newPgxInserter(conn pgxConn, cache MetricCache) (*pgxInserter, error) {
+	cmc := make(chan bool, 1)
+
+	inserter := &pgxInserter{
+		conn:                   conn,
+		metricTableNames:       cache,
+		completeMetricCreation: cmc,
 	}
+	//on startup run a completeMetricCreation to recover any potentially
+	//incomplete metric
+	err := inserter.CompleteMetricCreation()
+	if err != nil {
+		return nil, err
+	}
+
+	go inserter.runCompleteMetricCreationWorker()
+
+	return inserter, nil
 }
 
 type pgxInserter struct {
-	conn             pgxConn
-	metricTableNames MetricCache
-	inserters        sync.Map
+	conn                   pgxConn
+	metricTableNames       MetricCache
+	inserters              sync.Map
+	completeMetricCreation chan bool
```

Comment:
```suggestion
	completeMetricCreation chan struct{}
```
since we never use the value, we might as well. It indicates that it's just a signaling channel, and saves a bit of space.